content/2017.03.building-projects-as-a-way-to-learn.rst

```diff
@@ -0,0 +1,222 @@
+
+
+How do you turn this vision into OKRs?
+======================================
+
+OKR means **Objectives and Key results**: it is a way to define what
+you want to achieve in a tangible way and what you will be able to
+mesure
+`to see how effective you were <https://blog.betterworks.com/okrs-stretch-or-not-to-stretch/>`_.
+
+It is also a way to cut your vision in steps and to move your vision
+forward.
+
+
+How do you set course for a vision?
+===================================
+
+To be able to define OKRs from your vision, you need to be able to find
+the fastest way to get closer to your aim, however what we can learn
+from sailing is that the fastest way is rarely the shortest.
+
+At Mozilla, we have a vision that is written down in the
+`Mozilla Manifesto <https://www.mozilla.org/en-US/about/manifesto/>`_.
+
+Mitchell Baker does a great job making sure everyone at Mozilla stays
+aligned with that vision.
+
+However agreeing on the fastest way and turning the Mozilla vision
+into OKRs is really challenging and even more when you have a thousand
+people's opinion about it.
+
+It's the board's job to set course every six months (or even less) on
+what we should be focusing on for the next six months.
+
+When this happens, it often means changing our course and killing
+projects or at least putting them on the side if teams are turning to
+new priorities.
+
+At our level, we are driven by the use case. We have a vision of what
+we want to build and we keep adding ideas in the backlog.
+
+However we are not building anything until we have a proper use case
+for it. As soon as we have a use case it is added to our Roadmap.
+
+You endup not spending time on tasks you don't need and for each use
+case you are getting closer to your goal even if it might not be the
+way you'd taken initially.
+
+You also don't have to support code that you don't use.
+
+We even ended up removing features that were not used anymore.
+
+
+Our luck is opensource
+======================
+
+The advantage we have at Mozilla, is that everything we do is
+published as Free, Libre, Opensource Software, which means that, even
+if, as Mozilla employees, we focus on something else, the work we've
+done is still available for the community to care about it.
+
+Thunderbird is an example of this. Mozilla stopped spending resources
+on it in September 2007 but ten years later Mozilla Thunderbird is
+still actively used by a large number of users every day.
+
+When building software we solve complex issues and all the solutions
+we have put together are really useful for other people who might need
+to solve similar problems.
+
+- Mozilla Persona lead the ground for Firefox Accounts and the Portier
+  project.
+- The Firefox Hello loop server is a good example of how to build a
+  WebRTC signaling server.
+- The RemoteServiceWorker project shows how to use Python 3 asyncio
+  to handle a large pool of websockets using redis to route messages
+  between them.
```

Comment:
My point is that because it is open source the community is learning from it when they need to do a similar thing (like using WebRTC and their software) they can see how we did the signaling because they can read the code of the loop server for instance.